builtin/providers/aws/resource_aws_elasticache_replication_group.go

```diff
@@ -215,6 +220,8 @@ func resourceAwsElasticacheReplicationGroupRead(d *schema.ResourceData, meta int
 		d.Set("snapshot_window", c.SnapshotWindow)
 		d.Set("snapshot_retention_limit", c.SnapshotRetentionLimit)
 
+		d.Set("primary_endpoint_address", rgp.NodeGroups[0].PrimaryEndpoint.Address)
```

Comment:
As long as we have gotten into this block, we have passed a check to make sure that there is a NodeGroup available :) The first NodeGroup is the primary and we use that to populate all of the information above it